docs/examples/plot_spectral_harmonics.py

```diff
@@ -0,0 +1,166 @@
+# to 300 Hz.
+
+f0, voicing, voicing_probability = librosa.pyin(y=y, sr=sr, fmin=50, fmax=300)
+
+####################################################################
+# We can visualize the f0 contour on top of a spectrogram as follows
+
+S = np.abs(librosa.stft(y))
+
+times = librosa.times_like(S, sr=sr)
+
+fig, ax = plt.subplots()
+librosa.display.specshow(librosa.amplitude_to_db(S, ref=np.max),
+                         y_axis='log', x_axis='time', ax=ax)
+ax.plot(times, f0, linewidth=2, color='white', label='f0')
+ax.legend()
+
+##########################################################
+# The figure above illustrates how the f0 contour tends to
+# follow the lowest frequency with the most energy, which
+# are indicated by bright colors toward the bottom of the
+# image.  The patterns replicate at higher frequencies
+# corresponding to harmonics of the fundamental, which are
+# identified by `2*f0`, `3*f0`, `4*f0`, etc.
+#
+# We can use `librosa.f0_harmonics` to extract the energy
+# from a specified set of harmonics relative to the f0.
+
+# Let's use the first 30 harmonics: 1, 2, 3, ..., 30
+harmonics = np.arange(1, 31)
+
+# And standard Fourier transform frequencies
+frequencies = librosa.fft_frequencies(sr=sr)
+
+harmonic_energy = librosa.f0_harmonics(S, f0=f0, harmonics=harmonics, freqs=frequencies)
+
+#############################################################
+# We can visualize the result `harmonic_energy` alongside the
+# full spectrogram `S` to see how they compare to each other.
+
+# sphinx_gallery_thumbnail_number = 2
+
+fig, ax = plt.subplots(nrows=2, sharex=True)
+
+librosa.display.specshow(librosa.amplitude_to_db(S, ref=np.max),
+                         y_axis='log', x_axis='time', ax=ax[0])
+librosa.display.specshow(librosa.amplitude_to_db(harmonic_energy, ref=np.max),
+                         x_axis='time', ax=ax[1])
+ax[0].label_outer()
+ax[1].set(ylabel='Harmonics')
+
+#########################################################
+# In the above figure, we can observe the same general
+# patterns of spectral energy as in the full spectrogram,
+# but notice that the shapes no longer move up and down
+# with the f0 contour.  The resulting `harmonic_energy`
+# has been normalized with regard to the fundamental.
+#
+
+##############################################################
+# From the f0 contour and the harmonic energy measurements,
+# we can reconstruct an approximation to the original signal
+# by sinusoidal modeling.  This really just means adding up
+# sine waves at the chosen set of frequencies and scaled
+# appropriately by the amount of energy at that frequency
+# over time.
+#
+# Since the f0 contour is a time-varying frequency measurement,
+# the synthesis will need to support variable frequencies.
+# Luckily, the `mir_eval.sonify` module does exactly this.
+# We'll generate a separate signal for each harmonic separately,
+# and add them into a mixture to better approximate the original
+# signal.
```

Comment:
not sure why this is on its own line

Reply:
It doesn't need to be, but it doesn't hurt anything either.  It's kind of a quirk of how sphinx-gallery infers notebook / markdown from python comments, using `###...` barriers.
It looks best if the text under the barrier does not overflow, but the barrier length itself is arbitrary.

Reply:
i understand! ok